src/ioctls/device.rs

```diff
@@ -207,5 +227,18 @@ mod tests {
 
         assert!(device_fd.has_device_attr(&dist_attr).is_ok());
         assert!(device_fd.set_device_attr(&dist_attr).is_ok());
+
+        // Test `get_device_attr`. Here we try to extract the maximum supported number of IRQs.
+        // This value should be saved in the address provided to the ioctl.
+        let mut data: u32 = 0;
+
+        let mut gic_attr_get = kvm_bindings::kvm_device_attr::default();
+        gic_attr_get.group = KVM_DEV_ARM_VGIC_GRP_NR_IRQS;
+        gic_attr_get.addr = &mut data as *const u32 as u64;
```

Comment:
I would also exercise the error path for this. For example, `EFAULT` will be returned if the attribute points to an invalid address. So, I'd add this before checking the success case:
```
gic_attr_get.addr = data as u64;
// Without properly providing the address to where the 
// value will be stored, the ioctl fails with EFAULT.
let res = device_fd.get_device_attr(&mut gic_attr_get);
assert!(res.is_err() && errno::Error::last().errno() == 14);

gic_attr_get.addr = &mut data as *const u32 as u64;
...
...
```
Expect that coverage may need adjustments .

Reply:
Ahh yes yes this is a really good example. Thanks for proposing this! Updated.